### PR TITLE
usability improvements

### PR DESCRIPTION
**Pull request by bipinkrish**
https://github.com/sign-language-processing/pose/issues/116

### CODE REVIEW

File: src/js/pose_viewer/src/components/pose-viewer/pose-viewer.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import {Component, Element, Event, EventEmitter, h, Host, Method, Prop, State, Watch} from '@stencil/core';
 import type {PoseModel} from "pose-format/dist/types";
 import {Pose} from "pose-format";
+import {Buffer} from "buffer";
```
Comment (AmitMY) [inline src/js/pose_viewer/src/components/pose-viewer/pose-viewer.tsx:5]: i would prefer an `import type {Buffer}` so buffer does not transpile to javascript, as it is not available in all environments

File: src/js/pose_viewer/src/components/pose-viewer/pose-viewer.tsx [hunk 2]
```diff
@@ -86,14 +87,30 @@ export class PoseViewer {
...
-    // Abort previous request
+  private async loadPose() {
+    // Abort previous request if it exists
     if (this.fetchAbortController) {
       this.fetchAbortController.abort();
     }
 
-    this.fetchAbortController = new AbortController();
-    this.pose = await Pose.fromRemote(this.src, this.fetchAbortController);
+    if (this.src instanceof Buffer) {
```
Comment (AmitMY) [inline src/js/pose_viewer/src/components/pose-viewer/pose-viewer.tsx:96]: here, maybe a small refactor:

```ts
    if(typeof this.src === 'string') {
      const isRemoteUrl = this.src.startsWith('http') || this.src.startsWith('//');
      const isBrowserEnvironment = typeof window !== 'undefined';

      if (isRemoteUrl || isBrowserEnvironment) {
        // Remote URL or Browser environment
        this.fetchAbortController = new AbortController();
        this.pose = await Pose.fromRemote(this.src, this.fetchAbortController);
      } else {
        // Local environment
        this.pose = await Pose.fromLocal(this.src);
      }
    } else {
      this.pose = Pose.from(this.src);
    }
```

That moves the responsibility for error/type checking to the `from` method,

**Review by AmitMY — CHANGES_REQUESTED** [review]
minor changes

**Comment by bipinkrish** [comment]
should be okay now

**Review by AmitMY — APPROVED** [review]

**Comment by AmitMY** [comment]
Thanks!

**Pull request merged**